app/controllers/admin/commercials_controller.rb

```diff
@@ -57,7 +57,9 @@ def render_commercial
     def mass_upload
       errors = Commercial.read_file(params[:file]) if params[:file]
 
-      if errors.all? { |_k, v| v.blank? }
+      if errors.nil?
```

Comment:
I would say it would be better to separate the par without file that the one with file. So:

```ruby
if params[:file].blank
  flash[:error] = 'No file detected. Please attach a file.'
  return
end

if errors.all? { |_k, v| v.blank? }
 ...
```